src/DynamoRevit/DynamoRevit.cs

```diff
@@ -890,4 +912,85 @@ public override IEnumerable<string> GetDynamoUserDataLocations()
             return paths;
         }
     }
+
+    [Serializable()]
+    public class ExtensionInfo
+    {
+        public string NodePath { get; set; }
+        public string LayoutSpecsPath { get; set; }
+    }
+    internal static class DynamoRevitInternalNodes
+    {
+        private static IEnumerable<string> GetAllExtensionFiles()
+        {
+            string currentAssemblyPath = Assembly.GetExecutingAssembly().Location;
+            string currentAssemblyDir = Path.GetDirectoryName(currentAssemblyPath);
+
+            string internalNodesDir = Path.Combine(currentAssemblyDir, "packages");
+            if (false == Directory.Exists(internalNodesDir))
+            {
+                return new List<string>();
+            }
+
+            string[] internalNodesFolders = Directory.GetDirectories(internalNodesDir);
+
+            List<string> extensionFiles = new List<string>();
+            foreach (string dir in internalNodesFolders)
+            {
+                string extensionFile = Path.Combine(dir, "extension.xml");
```

Comment:
again, extension seems confusing given the existing and completely orthogonal extension functionality.

Reply:
renamed it